File: frontend/src/types/task.ts
export type ScaleType = 'HAMD' | 'HAMA' | 'PHQ-9'
export type TaskStatus = 'PENDING' | 'RUNNING' | 'COMPLETED' | 'FAILED'
export type TaskStage = 'ASR' | 'QC'
export type ReviewStatus = 'UNREVIEWED' | 'REVIEWED'

export interface QcResult {
  id: number
  doctor_score: number | null
  ai_score: number | null
  scoring_basis: string | null
  evidence_analysis: string | null
  error_reason: string | null
  optimization_suggestion: string | null
  item_results: QcItemResult[] | null
}

export interface QcItemResult {
  hamd_item: string
  doctor_score: number | null
  ai_score: number | null
  ai_scoring_basis: string | null
  difference?: number | null
  review_score?: number | null
  review_opinion?: string | null
}

export interface ReviewRecord {
  id: number
  reviewed_score: number
  review_reason: string
  review_comment: string | null
  reviewed_at: string
}

export interface TaskListItem {
  id: number
  scale_type: ScaleType
  audio_original_name: string
  audio_mime_type: string | null
  audio_size: number
  status: TaskStatus
  stage: TaskStage | null
  review_status: ReviewStatus
  error_message: string | null
  asr_text: string | null
  doctor_score: number | null
  ai_score: number | null
  created_at: string
  updated_at: string
}

export interface TaskDetail extends TaskListItem {
  qc_result: QcResult | null
  review_record: ReviewRecord | null
}

export interface TaskListResponse {
  items: TaskListItem[]
  total: number
  page: number
  page_size: number
}

File: backend/app/schemas/task.py
from datetime import datetime

from pydantic import BaseModel, Field

from app.models.assessment import ReviewStatus, ScaleType, TaskStage, TaskStatus


class QcResultResponse(BaseModel):
    id: int
    doctor_score: float | None
    ai_score: float | None
    scoring_basis: str | None
    evidence_analysis: str | None
    error_reason: str | None
    optimization_suggestion: str | None
    item_results: list[dict] | None = None

    model_config = {"from_attributes": True}


class ReviewRecordResponse(BaseModel):
    id: int
    reviewed_score: float
    review_reason: str
    review_comment: str | None
    reviewed_at: datetime

    model_config = {"from_attributes": True}


class TaskListItem(BaseModel):
    id: int
    scale_type: ScaleType
    audio_original_name: str
    audio_mime_type: str | None
    audio_size: int
    status: TaskStatus
    stage: TaskStage | None
    review_status: ReviewStatus
    error_message: str | None
    asr_text: str | None
    doctor_score: float | None = None
    ai_score: float | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class TaskDetailResponse(TaskListItem):
    qc_result: QcResultResponse | None = None
    review_record: ReviewRecordResponse | None = None


class TaskListResponse(BaseModel):
    items: list[TaskListItem]
    total: int
    page: int
    page_size: int


class ReviewCreate(BaseModel):
    reviewed_score: float = Field(..., ge=0)
    review_reason: str = Field(..., min_length=1, max_length=2000)
    review_comment: str | None = Field(default=None, max_length=5000)


class ItemReviewCreate(BaseModel):
    review_score: float = Field(..., ge=0)
    review_opinion: str = Field(..., min_length=1, max_length=2000)
